components/collector/src/source_collectors/npm/dependencies.py

```diff
@@ -1,24 +1,45 @@
 """npm dependencies collector."""
 
-from typing import cast
+from typing import TypedDict, cast
 
 from base_collectors import JSONFileSourceCollector
 from collector_utilities.type import JSON, JSONDict
 from model import Entities, Entity
 
 
+class DependencyVersionInfo(TypedDict):
+    """Dependency version information."""
+
+    current: str  # Current version
+    dependent: str  # Dependent component, i.e. user of the dependency
+    latest: str  # Latest version
+    location: str  # Location of the package
+    wanted: str  # Wanted version
+
+
 class NpmDependencies(JSONFileSourceCollector):
     """npm collector for dependencies."""
 
     def _parse_json(self, json: JSON, filename: str) -> Entities:
         """Override to parse the dependencies from the JSON."""
-        return Entities(
-            Entity(
-                key=f'{dependency}@{versions.get("current", "?")}',
-                name=dependency,
-                current=versions.get("current", "unknown"),
-                wanted=versions.get("wanted", "unknown"),
-                latest=versions.get("latest", "unknown"),
-            )
-            for dependency, versions in cast(JSONDict, json).items()
+        entities = Entities()
+        for dependency, version_info in cast(JSONDict, json).items():
+            versions = version_info if isinstance(version_info, list) else [version_info]
+            for version in versions:
+                entities.append(self._create_entity(dependency, version))
+        return entities
+
+    def _create_entity(self, dependency: str, version: DependencyVersionInfo) -> Entity:
+        """Create an entity for the dependency."""
+        key = f'{dependency}@{version.get("current", "?")}'
+        name = dependency
+        if "dependent" in version:
+            key += f"@{version['dependent']}"
```

Comment:
So we could potentially have a key like `package@?@dep`, is that problematic?

Reply:
I don't think so. A dependency should be uniquely identified by package name, current version ("?" if unknown), and dependent. The key is not visible to the users, so aesthetics do not come into play :-)